llvm-external-projects/iree-dialects/lib/Dialect/VectorExt/IR/VectorExtDialect.cpp

```diff
@@ -153,3 +153,21 @@ std::optional<int64_t> LayoutAttr::getBatchDim(int64_t dim) {
   }
   return std::nullopt;
 }
+
+std::tuple<int64_t, int64_t, int64_t> LayoutAttr::getLaneGrid() {
+  int64_t laneX = 1;
+  int64_t laneY = 1;
+  int64_t laneZ = 1;
+  for (PerDimLayoutAttr dimLayout : getLayouts()) {
+    auto maybeXShape = dimLayout.getShape(LayoutDimension::LANEX);
```

Comment:
What do you think of laneX *= dimLayout.getShape(LayoutDimension::LANEX).value_or(1) here and below? 
Saves the if(maybeXShape) ?

Reply:
Thanks, this looks cleaner

Reply:
Also added a comment to indicate that it isn't doing an accumulation for when someone else looks at it.